lib/addressable/uri.rb

```diff
@@ -144,9 +144,7 @@ def self.parse(uri)
           /:([^:@\[\]]*?)$/, EMPTY_STR
         )
         port = authority[/:([^:@\[\]]*?)$/, 1]
-      end
-      if port == EMPTY_STR
-        port = nil
+        port = nil if port == EMPTY_STR
```

Comment:
Looks like this assignment was moved inside `if authority != nil` where it wasn't before. 
Not sure it matters (all tests still pass, but we might lack coverage).

Spotted this when looking at the files instead of the diff.

### This PR

https://github.com/sporkmonger/addressable/blob/7379545925e2a994dadb6a21a5f9a12051a115fd/lib/addressable/uri.rb#L134-L148

### `main`

https://github.com/sporkmonger/addressable/blob/8657465b438d93c707aff4346fdcf74c742b4a53/lib/addressable/uri.rb#L134-L150

Reply:
correct.  port is set to `nil` on line 133 before the block, so this check only needs to happen if the authority block changes the port value

Reply:
added a test to ensure proper functionality